08-persistence-and-orm.ipynb

```diff
@@ -652,6 +652,32 @@
     "courseWithSubject.Display();"
    ]
   },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": [
+    "### Materialization with Entity Framework  \n",
+    "\n",
+    "Materialization is the process where data retrieved from a database is converted into .NET objects. In Entity Framework, this process is handled automatically when queries are executed. Data is mapped to entity properties, and relationships are resolved based on configurations.  \n",
+    "\n",
+    "Eager loading fetches related data together with the main entity. It is defined in the query and is preferred when related data is required immediately to avoid additional database calls.  \n",
+    "\n",
+    "Explicit loading retrieves related data later, after the main entity has been loaded. It allows more control over when data is fetched but may result in additional queries.  \n",
```

Comment:
Not entirely sure if it results in additional queries. AFAIK just more `JOIN` statements. Would be worth an example if that is the case.